plugins/modules/notification/discord.py

```diff
@@ -0,0 +1,183 @@
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+    content: "This is a message from ansible"
+    username: Ansible
+    avatar_url: "https://docs.ansible.com/ansible/latest/_static/images/logo_invert.png"
+
+- name: Send a embedded message to the discord channel
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+    embeds:
+      - title: "Embedded message"
+        description: "This is an embedded message"
+        footer:
+          text: "Author: Ansible"
+        image:
+          url: "https://docs.ansible.com/ansible/latest/_static/images/logo_invert.png"
+"""
+
+RETURN = """
+http_code:
+  description:
+    - Response Code returned by discord API.
+  returned: always
+  type: int
+  sample: 204
+"""
+
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils.basic import AnsibleModule
+
+
+def discord_check_mode(module):
+
+    webhook_id = module.params['webhook_id']
+    webhook_token = module.params['webhook_token']
+
+    headers = {
+        'content-type': 'application/json'
+    }
+
+    url = "https://discord.com/api/webhooks/%s/%s" % (
+        webhook_id, webhook_token)
+
+    response, info = fetch_url(module, url, method='GET', headers=headers)
+    return response, info
+
+
+def discord_text_msg(module):
+
+    webhook_id = module.params['webhook_id']
+    webhook_token = module.params['webhook_token']
+    content = module.params['content']
+    user = module.params['username']
+    avatar_url = module.params['avatar_url']
+    tts = module.params['tts']
+    embeds = module.params['embeds']
+
+    headers = {
+        'content-type': 'application/json'
+    }
+
+    url = "https://discord.com/api/webhooks/%s/%s" % (
+        webhook_id, webhook_token)
+
+    payload = {
+        'content': content,
+        'username': user,
+        'avatar_url': avatar_url,
+        'tts': tts,
+        'embeds': embeds,
+    }
+
+    payload = module.jsonify(payload)
+
+    response, info = fetch_url(module, url, data=payload, headers=headers)
+    return response, info
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            webhook_id=dict(type='str', required=True),
+            webhook_token=dict(type='str', required=True, no_log=True),
+            content=dict(type='str'),
+            username=dict(type='str'),
+            avatar_url=dict(type='str'),
+            tts=dict(type='bool', default=False),
+            embeds=dict(type='list', elements='dict'),
+        ),
+        required_one_of=[['content', 'embeds']],
+        supports_check_mode=True
+    )
+
+    result = dict(
+        changed=False,
+        http_code='',
+    )
+
+    if module.check_mode:
+        response, info = discord_check_mode(module)
+        if info['status'] != 200:
+            module.fail_json(http_code=info['status'], msg=info['msg'], response=module.from_json(info['body']), info=info)
```

Comment:
```suggestion
            module.fail_json(http_code=info['status'], msg=info['msg'], response=module.from_json(info['body']), info=info)
```
`'body'` is not necessarily in `info`. (Same below.)

Reply:
Does it make no sense to format der json response as output ?

Reply:
It does, but this line can result in a `KeyError` if `'body' not in info` (or some other error when `modlue.from_json()` fails).

Reply:
Yeah, that's right. Will remove this.

Reply:
Instead of removing it, you could also add `try`/`except` to handle errors gracefully.